### PR TITLE
Updating to use main branch for sandman

### PR DESCRIPTION
**Pull request by paigerube14**
Adding to include sandman in the mian branch and easily refernce variables in artifacts

Passing job: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/paige-e2e-multibranch/job/write-to_sheet/139/

Prow stuff isnt hurting anything

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+gspread
+oauth2client
+datetime
+pytz
+pyyaml
```
Comment (nathan-weinberg) [inline requirements.txt:5]: Is this file for `sandman.py`?

File: write_to_sheet/write_scale_results_sheet.py [hunk 2]
```diff
@@ -177,3 +179,82 @@ def write_to_sheet(google_sheet_account, flexy_id, ci_job, job_type, job_url, st
...
+        print('get latency params ' + str(creation_time) )
+        print('get latency params ' + str(es_username) )
+        print('get latency params ' + str(es_password) )
+        row.extend(write_helper.get_pod_latencies(uuid, creation_time, es_username,es_password))
+
+    row.append(str(datetime.now(tz)))
+    ws = sheet.worksheet(job_type)
+    ws.insert_row(row, index, "USER_ENTERED")
+
+
```
Comment (nathan-weinberg) [inline write_to_sheet/write_scale_results_sheet.py:260]: Remove extra newline

File: write_to_sheet/write_scale_results_sheet.py [hunk 3]
```diff
@@ -177,3 +179,82 @@ def write_to_sheet(google_sheet_account, flexy_id, ci_job, job_type, job_url, st
     row.append(str(write_helper.get_env_vars_from_file(env_vars_file)))
     row.append(user)
     ws.insert_row(row, index, "USER_ENTERED")
+
+def write_prow_results_to_sheet():
+    scopes = [
+    'https://www.googleapis.com/auth/spreadsheets',
+    'https://www.googleapis.com/auth/drive'
+    ]
```
Comment (nathan-weinberg) [inline write_to_sheet/write_scale_results_sheet.py:187]:
```suggestion
    scopes = [
        'https://www.googleapis.com/auth/spreadsheets',
        'https://www.googleapis.com/auth/drive'
    ]
```

File: write_to_sheet/write_scale_results_sheet.py [hunk 4]
```diff
@@ -177,3 +179,82 @@ def write_to_sheet(google_sheet_account, flexy_id, ci_job, job_type, job_url, st
...
+    google_sheet_account = os.getenv("GSHEET_KEY_LOCATION")
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(google_sheet_account, scopes) #access the json key you downloaded earlier
+    file = gspread.authorize(credentials) # authenticate the JSON key with gspread
+    sheet = file.open_by_url("https://docs.google.com/spreadsheets/d/1cciTazgmvoD0YBdMuIQBRxyJnblVuczgbBL5xC8uGuI/edit?usp=sharing")
```
Comment (nathan-weinberg) [inline write_to_sheet/write_scale_results_sheet.py:191]: Perhaps the URL here should be a constant set at the top of the file? Something like 
```python
WRITE_SHEET_URL = "https://docs.google.com/spreadsheets/d/1cciTazgmvoD0YBdMuIQBRxyJnblVuczgbBL5xC8uGuI/edit?usp=sharing"
...
sheet_obj = file.open_by_url(WRITE_SHEET_URL)
```

File: write_to_sheet/write_scale_results_sheet.py [hunk 5]
```diff
@@ -177,3 +179,82 @@ def write_to_sheet(google_sheet_account, flexy_id, ci_job, job_type, job_url, st
...
+    #open sheet
+    print('hello from prow write to sheet')
+    index = 2
+    ws = sheet.worksheet("test")
+    ws.insert_row(["hello"], index, "USER_ENTERED")
```
Comment (nathan-weinberg) [inline write_to_sheet/write_scale_results_sheet.py:196]: Is this testing code or meant for merging?

File: write_to_sheet/write_scale_results_sheet.py [hunk 6]
```diff
@@ -177,3 +179,82 @@ def write_to_sheet(google_sheet_account, flexy_id, ci_job, job_type, job_url, st
...
+        return_code, CLUSTER_NAME=write_helper.run("oc get machineset -n openshift-machine-api -o=go-template='{{(index (index .items 0).metadata.labels \"machine.openshift.io/cluster-api-cluster\" )}}'")
+        start_time = parse_output_for_starttime()
+        if return_code == 0:
+            grafana_cell = find_uperf_uuid_url(CLUSTER_NAME,start_time,es_username,es_password)
+        else:
+            grafana_cell = ""
+    elif job_type == "router-perf":
+            global uuid
+            uuid, metadata = get_router_perf_uuid()
+            grafana_cell = uuid
```
Comment (nathan-weinberg) [inline write_to_sheet/write_scale_results_sheet.py:212]:
```suggestion
    elif job_type == "router-perf":
        global uuid
        uuid, metadata = get_router_perf_uuid()
        grafana_cell = uuid
```

File: Jenkinsfile [hunk 7]
```diff
@@ -105,20 +108,26 @@ pipeline {
...
-                env.UUID = sh(returnStdout: true, script: "jq -r '.uuid' $WORKSPACE/ocp-qe-perfscale-ci-netobs/data/workload.json").trim()
+                buildInfo = readJSON file: 'help_scripts/data/workload.json'
+                println "build info $buildInfo"
+                buildInfo.each { env.setProperty(it.key.toUpperCase(), it.value) }
+                currentBuild.description = "Write to sheet sandman info: <br/>"
+                archiveArtifacts(
+                    artifacts: 'help_scripts/data/*',
+                    allowEmptyArchive: true,
+                    fingerprint: true
+                )
```
[the remark this answers is not in the document]
Comment (paigerube14) [reply Jenkinsfile]: nope took out now!

File: requirements.txt (hunk 1 above)
Comment (paigerube14) [reply requirements.txt:5]: This is a file to help install the requirements that are needed in the write results part not sandman (moved to a file for prow integration)

File: write_to_sheet/write_scale_results_sheet.py (hunk 5 above)
Comment (paigerube14) [reply write_to_sheet/write_scale_results_sheet.py:196]: This part is a work in progress. Will move to another branch

File: requirements.txt (hunk 1 above)
Comment (paigerube14) [reply requirements.txt:5]: Updated to use this in jenkins as well

**Comment by paigerube14** [comment]
Working job here: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/scale-ci/job/paige-e2e-multibranch/job/write-to_sheet/145/console

File: Jenkinsfile [hunk 8]
```diff
@@ -103,22 +106,24 @@ pipeline {
...
                     println 'Successfully ran Mr. Sandman tool :)'
                 }
-                // update build description fields
-                // UUID
-                currentBuild.description = "Write to sheet sandman info: "
-                env.UUID = sh(returnStdout: true, script: "jq -r '.uuid' $WORKSPACE/ocp-qe-perfscale-ci-netobs/data/workload.json").trim()
+                // update build description fields 
+                buildInfo = readJSON file: 'help_scripts/data/workload.json'
+
+                // Add enviornment variables of sandman values
```
Comment (nathan-weinberg) [inline Jenkinsfile:112]:
```suggestion
                // Add environment variables of sandman values
```

File: Jenkinsfile [hunk 9]
```diff
@@ -103,22 +106,24 @@ pipeline {
                 else {
                     println 'Successfully ran Mr. Sandman tool :)'
                 }
-                // update build description fields
-                // UUID
-                currentBuild.description = "Write to sheet sandman info: "
-                env.UUID = sh(returnStdout: true, script: "jq -r '.uuid' $WORKSPACE/ocp-qe-perfscale-ci-netobs/data/workload.json").trim()
+                // update build description fields 
+                buildInfo = readJSON file: 'help_scripts/data/workload.json'
```
Comment (nathan-weinberg) [inline Jenkinsfile:110]: Do we want this to be `buildInfo` or something like `workloadInfo` or something else?

**Review by nathan-weinberg — COMMENTED** [review]
Couple small comments but otherwise LGTM!

File: write_to_sheet/write_scale_results_sheet.py (hunk 4 above)
Comment (nathan-weinberg) [reply write_to_sheet/write_scale_results_sheet.py]: @paigerube14 not sure if this comment is still relevant or not, anyway just an opinion won't block merge on it :+1:

File: Jenkinsfile (hunk 8 above)
Comment (paigerube14) [reply Jenkinsfile:112]: I have the hardest time spelling that word!!!

File: write_to_sheet/write_scale_results_sheet.py (hunk 4 above)
Comment (paigerube14) [reply write_to_sheet/write_scale_results_sheet.py]: i find it easier to find in line.. will leave for now

**Pull request merged**